lib/rules/require-padding-newlines-in-files.js

```diff
@@ -0,0 +1,53 @@
+ /**
+ * Requires the file to start and/or end with a newline.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values: `true`, `"beginningOfFileOnly"` or `"endOfFileOnly"`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requirePaddingNewlinesInFiles": true
+ * "requirePaddingNewlinesInFiles": "beginningOfFileOnly"
+ * "requirePaddingNewlinesInFiles": "endOfFileOnly"
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(options) {
+        assert(
+            options === true || options === 'beginningOfFileOnly' || options === 'endOfFileOnly',
+            this.getOptionName() + ' option requires value to be `true`, `beginningOfFileOnly` or `endOfFileOnly`'
+        );
+
+        if (options === 'beginningOfFileOnly') {
+            this._atBOF = true;
+            this._atEOF = false;
+        } else if (options === 'endOfFileOnly') {
+            this._atBOF = false;
+            this._atEOF = true;
+        } else {
+            this._atBOF = true;
+            this._atEOF = true;
+        }
+    },
+
+    getOptionName: function() {
+        return 'requirePaddingNewlinesInFiles';
+    },
+
+    check: function(file, errors) {
+        if (this._atBOF && file.getLines()[0] !== '') {
+            errors.add('Missing padding new line at beginning of file', 1, 0);
+        }
+
+        if (this._atEOF && file.getLines()[file.getLines().length - 2] !== '') {
```

Comment:
Good question, but this depends on `requireLineFeedAtFileEnd` :-/

If `requireLineFeedAtFileEnd` is set to `true`, this will pass:

```
// ←this line by BOF
var x;
// ←this line by EOF
// ←this is the line feed
```

If not, this will pass:

```
// ←this line by BOF
var x;
// ←this line by EOF, *and* it is also the line feed
```

Forgetting the BOF rule (ie using `endOfFileOnly`), this will fail if `requireLineFeedAtFileEnd` is set:
```
var x;
// ←New line, which is the line feed
```

Think of it in the opposite rule (I should have started with it actually). If you set `disallowPaddingNewlinesInFiles` to `true`, you want your files to be:

```js
var x;
```

But if you **also** set `requireLineFeedAtFileEnd` to `true`, then only files with that format must pass:

```js
var x;
// Line feed, empty line
```

Although the last line is empty, it respects `requireLineFeedAtFileEnd` (the line feed being often used by systems such as git) **and** `disallowPaddingNewlinesInFiles` (as both lines at top and bottom are style choices this time).

Does that make sense?
It seems like I have to interact with another rule (know if it is set), is there a way to do that? If not, do you see an alternative use case we could choose to avoid this issue?